src/Microsoft.Health.Fhir.SqlServer/Features/Search/CustomQueries.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Core;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    internal static class CustomQueries
+    {
+        private static DateTimeOffset _lastUpdatedQueryCache = DateTimeOffset.MinValue;
+        private static object lockObject = new object();
+        public static readonly ConcurrentDictionary<string, string> QueryStore = new ConcurrentDictionary<string, string>();
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=31&length=20)

If you build entire dictionary and swap, there is no need to have it concurrent.

Reply:
I now simply swap the dictionary.